airbyte-integrations/connectors/source-bing-ads/source_bing_ads/manifest.yaml

```diff
@@ -1630,6 +1923,156 @@ definitions:
           {% set v = record.get('TopImpressionRatePercent') %}
           {{ v | replace('%', '') | float if v is not none and v != "" else None }}
 
+    # used in hourly report streams to convert date-time format
+    hourly_stream_timestamp_transformation:
+      - type: AddedFieldDefinition
+        path:
+          - TimePeriod
+        #            Bing Ads API reports with hourly aggregation provides date fields in custom format: "2023-11-04|11"
+        #            Return date in RFC3339 format: "2023-11-04T11:00:00+00:00"
+        value: |
+          {%- set parts = record.TimePeriod.split('|') -%}
+          {%- set time_period = record.TimePeriod -%}
+          {%- set date_part = '' -%}
+          {%- set hour_part = 0 -%}
+          {%- if parts|length > 1 -%}
+            {%- set date_part = parts[0] -%}
+            {%- set hour_part = parts[1]|int -%}
+            {%- set time_period = date_part ~ "T" ~ "%02d" % hour_part ~ ":00:00+00:00" -%}
+          {%- endif -%}
+          {{ time_period }}
+
+    # Used in campaign performance report streams
+    campaign_performance_report_common_transformations:
+      - type: AddedFieldDefinition
+        path:
+          - CampaignId
+        value: |
+          {{ record['CampaignId'] | int if record.get('CampaignId') else none }}
```

Comment:
I have added schema [normalization](https://github.com/airbytehq/airbyte/pull/61523/files#diff-e940c9c1127f2dfa6d01ef1e93b0ff7338d71d46fa225de7bd0d0aaf7cf2940aR1154) to avoid those transformations.